src/evaluation/Classes/WordHitAtK.py

```diff
@@ -0,0 +1,111 @@
+        :param model: model to be tested
+        :param data: data to be tested on. could be string if its a path to test json file or dict if its a ready to go
+                     list in form of [{"text": "...", "missing": {...}}]
+        :param k: the k of hit@k
+        :return: hit@k score (# of words hists at k)/(# of missing words)
+        """
+        if type(data) == 'str':
+            data = self.get_data_at_hit_at_k_test_format(data)
+
+        for entry_idx, entry in enumerate(data):
+            real_values = entry['missing']
+            modelRes = model.predict(entry['text']).get_only_k_predictions(k)
+            list_of_preds = self._model_result_to_list_of_preds(modelRes)
+            print(list_of_preds) 
+            all_words_and_missing_indexes = []
+            missing_idxs_full_word_mask = self._get_missing_idxs(entry['text']) #getting all the missing indexes that we want to predict (for words only !)
+            for pred_idx, preds in enumerate(list_of_preds):
+                pred_missing_words=[]
+                for j in range(len(preds)):
+
+                     #Because we return only the indexes from missing words and in the verse could be another missings (chars/subword but not full word)
+                     #So the length of the list of missing indexes words shorter than the predictions length. If it correct we finish the process and find all the
+                     #missing indexes and their predictions
+                      if pred_idx>len(missing_idxs_full_word_mask)-1:
+                        break
+
+                       #could be a situation that the model returns a shorter word than what we want to predict  
+                      if len(missing_idxs_full_word_mask[pred_idx])>len(preds[j]):
+                          pass
+                      else:
+                          all_words_and_missing_indexes.append((missing_idxs_full_word_mask[pred_idx],preds[j]))
+
+            #all_words_and_missing_indexes holds lists of tuples: [([4,5,6,7],'עוהב'),([4,5,6,7], 'אוהב')...]
+            fit_count = 0
+            for i, c_preds in enumerate(all_words_and_missing_indexes):
+                flag=True
+                for m,mis_index in enumerate(c_preds[0]):
+
+                  #if there is  different between one of the chars from the word we predict to the real word the flag will be false, else will be true
+                  # and add 1 to the amount of words that we corrected
+                  
+                  if real_values[str(mis_index)] != c_preds[1][m]:
+                     flag=False
+                
+                if flag==True:
+                  fit_count+=1
+            
+            print(all_words_and_missing_indexes)
+            if len(all_words_and_missing_indexes) ==0:
+              return 0
+            return fit_count / len(all_words_and_missing_indexes)
+
+    def _model_result_to_list_of_preds(self, modelRes: ModelResult) -> List[List[str]]:
```

Comment:
don't duplicate code.
please move it this to the parent class and use it properly. do so also for the other class that you copied it from